apps/order/serializers.py

```diff
@@ -24,13 +24,29 @@ def validate_book(self, book):
 
 class CreateOrderFromCartSerializer(CreatedUpdatedBaseSerializer, serializers.ModelSerializer):
 
+    cart_item_ids = serializers.ListField(
+        child=serializers.IntegerField(required=True)
+    )
+
     class Meta:
         model = Order
-        fields = ()
+        fields = ('cart_item_ids',)
+
+    def validate_cart_item_ids(self, cart_item_ids):
+        created_by = self.context['request'].user
+        cart_items = CartItem.objects.filter(
+            created_by=created_by, id__in=cart_item_ids
+        )
+        if not cart_items.exists():
```

Comment:
One of the cart_item existing is okay, not all?

Reply:
@thenav56  Improved validation logic

Reply:
Using len is enough since we already have the data.
```python3
        filtered_cart_items_ids = list(cart_items.values_list('id', flat=True))
        if len(filtered_cart_items_ids) != len(cart_item_ids):
            raise serializers.ValidationError(_('Invalid cart item id supplied.'))
````

Reply:
@thenav56  Removed exists part